test/framework/src/main/java/org/elasticsearch/snapshots/mockstore/MockRepository.java

```diff
@@ -250,6 +250,14 @@ public synchronized void unblock() {
         this.notifyAll();
     }
 
+    public void setRandomControlIOExceptionRate(double randomControlIOExceptionRate) {
+        this.randomControlIOExceptionRate = randomControlIOExceptionRate;
+    }
+
+    public void setRandomDataFileIOExceptionRate(double randomDataFileIOExceptionRate) {
+        this.randomDataFileIOExceptionRate = randomDataFileIOExceptionRate;
+    }
```

Comment:
May be it is worth logging something when overriding those values?
Those values are originally set via metadata and I suspect it might be confusing to see them suddenly become different.

Reply:
Hi @idegtiarenko ! There are already functions that can be used to change the settings without logging. E.g., `blockOnDataFiles` is set by a metadata (to e.g. false) and then there is `blockOnDataFiles()` that can set it. So I thought of just mimicking those. But I see your point, so I updated the PR and introduced debug logging to it.